compiler/modules/print.llvm/llvm.bal

```diff
@@ -577,6 +605,241 @@ public class FunctionDefn {
+    string[] preamble = [];
+    function init(string ref) {
+        self.label = ref;
+    }
+
+    function addLine(string ...words) {
+        self.lines.push(concat(...words));
+    }
+
+    function addPreamble(string ...words) {
+        self.preamble.push(concat(...words));
+    }
+
+    function resetPreamble() {
+        self.preamble = [];
+    }
+
+    function hasPreamble() returns boolean {
+        return self.preamble.length() != 0;
+    }
+
+    function output(Output out) {
+        if self.lines.length() == 0 {
+            out.push(string `${self.ref()} = !{}`);
+        }
+        foreach var line in self.lines {
+           out.push(line);
+        }
+    }
+
+    function outputPreamble(Output out) {
+        foreach var line in self.preamble {
+           out.push(line);
+        }
+    }
+
+    function ref() returns string {
+        return self.label;
+    }
+}
+
+final readonly & map<string> sourceLangToString = {
+    "C99": "DW_LANG_C99"
+};
+
+final readonly & map<string> emissionKindToString = {
+    "none": "NoDebug",
+    "full": "FullDebug",
+    "lineTablesOnly": "LineTablesOnly"
+};
+
+final readonly & map<string> moduleFlagBehaviorToString = {
+    "error": "1",
+    "warning": "2",
+    "require": "3",
+    "override": "4",
+    "append": "5",
+    "appendUnique": "6",
+    "max": "7"
+};
+
+# Corresponds to LLVMDIBuilderRef
+public class DIBuilder {
+    Metadata[] metadata = [];
+    Metadata[] moduleFlags = [];
+    Metadata? compileUnit = ();
+    function init(Module module) {}
+
+    function genMetadataLabel() returns string {
+        return string `!${self.metadata.length()}`;
+    }
+
+    function addMetadata() returns Metadata {
+        Metadata metadata = new(self.genMetadataLabel());
+        self.metadata.push(metadata);
+        return metadata;
+    }
+
+    // Corresponds to LLVMDIBuilderCreateCompileUnit
+    public function createCompileUnit(*CompileUnitProperties props) returns Metadata {
+        Metadata metadata = self.addMetadata();
+        Metadata? file = props.file;
+        string[] words = [];
+        words.push(metadata.ref(), "=", "distinct", "!", "DICompileUnit", "(", "language",
+                         ":", sourceLangToString.get(props.language), ",", "file", ":");
+        if file is Metadata {
+            words.push(file.ref());
+        }
+        else {
+            words.push("null");
+        }
+        words.push(",", "isOptimized", ":", props.isOptimized.toString(), "," , "runtimeVersion", ":",
+                   props.runtimeVersion.toString(), ",", "emissionKind", ":", emissionKindToString.get(props.kind), ")");
+        metadata.addLine(...words);
+
+        metadata.addPreamble("!", "llvm", ".", "dbg", ".", "cu", "=", "!", "{", metadata.ref(), "}");
+        self.compileUnit = metadata;
+        return metadata;
+    }
+
+    // Corresponds to LLVMDIBuilderCreateFile
+    public function createFile(string fileName, string directory) returns Metadata {
```

Comment:
filename is one word so use `filename` not `fileName`.

Reply:
Fixed